.github/workflows/build_appliances.yaml

```diff
@@ -12,9 +12,29 @@ jobs:
       run: |
         git clone https://www.github.com/bdunne/imagefactory.git /build/gha/$GITHUB_RUN_ID/imagefactory
   build_appliances:
+    strategy:
+      max-parallel: 2
+      matrix:
+        target:
+        - azure
+        - ec2
+        - gce
+        - hyperv
+        - libvirt
+        - openstack
+        - ovirt
+        - vagrant
+        - vsphere
     runs-on: self-hosted
     needs: setup
+    env:
+      TARGET_PLATFORM: ${{ matrix.target }}
     steps:
     - uses: actions/checkout@v3
     - name: Build appliance images
-      run: "bin/nightly-build.sh"
+      env:
+        IBM_CLOUD_ACCESS_KEY: ${{ secrets.IBM_CLOUD_ACCESS_KEY }}
+        IBM_CLOUD_ENDPOINT: ${{ secrets.IBM_CLOUD_ENDPOINT }}
+        IBM_CLOUD_SECRET_KEY: ${{ secrets.IBM_CLOUD_SECRET_KEY }}
```

Comment:
Naming wise, I think it's preferable to not say IBM_CLOUD, but something generic like BUILD_SYSTEM, or just BUILD, but I'm fine to merge without that change.